VCFv4.4.tex

```diff
@@ -522,8 +526,25 @@ \subsubsection{Genotype fields}
   All phased genotypes that do not contain a PS subfield are assumed to belong to the same phased set.
   If the genotype in the GT field is unphased, the corresponding PS field is ignored.
   The recommended convention is to use the position of the first variant in the set as the PS identifier (although this is not required).
-\end{itemize}
-
+  \item PQL (List of integers): The list of PQs, one for each phase set in PSL (encoded like PQ). The missing value 0 should be used when the corresponding PSL value is missing, or when the phasing is of unknown quality.
+  \item PSL (List of Strings): The list of phase sets, one for each allele specified in the {\tt GT}. 
+  Unphased alleles (without a $\mid$ separator after them) must have the value '$.$' in their corresponding position in the list. 
+  The phase-set name must be unique across the genome (unlike {\tt PS}  as this tag can be used to phase across references.
+  Thus, the recommended value for phase-set names is CHR*POS*ALLELE-NUMBER of the "first" allele which is included in this set, with ALLELE-NUMBER being the index of the allele in the {\tt GT} field, since multiple distinct phase-sets could start at the same position. \footnote{The '*' character is used as a separator since ':' is already a separator in VCF.}
+  A given sample-genotype must not have values for both PS and PSL.
+  In addition, PS and PSL are not interoperable, in that a PS mentioned in one variant cannot be referenced in a PSL in another, 
+  since when used in PS is isn't connected to any specific haplotype (i.e. first or second), but PSL is. 
+ 
+  Example: 
+  
+  \vspace{0.5em}
+  \begin{tabular}{ l l l l l l l l l l}
+	\#CHROM & POS & ID & REF & ALT & QUAL & FILTER & INFO & FORMAT & SAMPLE1\\
+	chr19 & $5$ & . & T & G & . & PASS & DP=100 &GT:PSL & \tt{0|1:.,chr9*5*1}\\
+	chr20 & $10$ & . & A & T,G & . & PASS & DP=100 &GT:PSL & \tt{1|2/3|:chr20*10*1,.,chr9*5*1} \\
```

Comment:
Using "1|2/3|" instead of "1|2/3" implies the binding of / and | is to the left.  However the BCF spec indicates it is to the right so you'd need the extra symbol at the start and not the end.

From the table on page 32 of https://github.com/samtools/hts-specs/blob/master/VCFv4.3.pdf it shows "0 / 1 | 2" as 0x02 04 07 via the formula `(allele+1)<<1 | phased`.  The third byte is the phased one,

This really doesn't seem to be clearly defined in the spec, and is causing confusion.  See https://github.com/samtools/htslib/issues/1113 for an example of this very problem.

Reply:
This is a good point and also the behavior of the reference implementation in htslib. I'd be open to modifying the existing specification and htslib since this mixed ploidy is not a widely used feature.

Reply:
there's no need...I agree that it's best to keep with the current example.